labtool2.0/src/components/TagLabel.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import { Button, Icon } from 'semantic-ui-react'
+import PropTypes from 'prop-types'
+
+export const TagLabel = props => {
+  const { color, text, handleClick, tag, basic, removeLabel, disabled } = props
+  const validColors = ['red', 'orange', 'yellow', 'olive', 'green', 'teal', 'blue', 'violet', 'purple', 'pink', 'brown', 'grey', 'black']
+
+  const tagColor = tag ? tag.color : color
+  const tagText = tag ? tag.name : text
+
+  const backgroundColor = backgroundColor => {
+    return `${validColors.includes(backgroundColor) ? '' : backgroundColor}`
+  }
+
+  const borderColor = backgroundColor => {
+    if (!basic) return ''
+    if (validColors.includes(backgroundColor)) return ''
+    return `1px solid ${backgroundColor}`
+  }
+
+  const hexToRgb = hex => {
+    hex = '0x' + hex.substring(1, 7)
+    let r = (hex >> 16) & 0xff
+    let g = (hex >> 8) & 0xff
+    let b = hex & 0xff
+    return [r, g, b]
+  }
+
+  /**
+   * If background color is hexcode, use this function to define text color.
+   * @param {*} backgroundColor
+   */
+  const textColor = backgroundColor => {
+    if (validColors.includes(backgroundColor)) return ''
+    if (basic) return 'black'
+
+    const rgb = hexToRgb(backgroundColor)
+    const rgbSum = Math.round((parseInt(rgb[0]) * 299 + parseInt(rgb[1]) * 587 + parseInt(rgb[2]) * 114) / 1000)
+    return rgbSum > 125 ? 'black' : 'white'
```

Comment:
125 is too low, there are some colors with which white would look better. Perhaps something like 140-150? It'll need some experimenting